templates/renew-certs.py

```diff
@@ -1,46 +1,97 @@
+    for cert in certs:
+        if os.access(cert['certpath'], os.F_OK):
+            stat = os.stat(cert['certpath'])
+            print 'Certificate file ' + cert['certpath'] + ' already exists'
 
-for cert in certs:
-    if os.access(cert['certpath'], os.F_OK):
-        stat = os.stat(cert['certpath'])
-        print "Certificate file " + cert['certpath'] + " already exists"
+            if time.time() - stat.st_mtime < {{ letsencrypt_min_renewal_age }} * 86400:
+                print ('  The certificate is younger than {{ letsencrypt_min_renewal_age }} days.' +
+                       ' Not creating a new certificate.\n')
+                continue
 
-        if time.time() - stat.st_mtime < {{ letsencrypt_min_renewal_age }} * 86400:
-            print "  The certificate is younger than {{ letsencrypt_min_renewal_age }} days. Not creating a new certificate.\n"
-            continue
+        host = ','.join(cert['host']) if type(cert['host']) is list else cert['host']
 
-    host = ",".join(cert['host']) if type(cert['host']) is list else cert['host']
+        print 'Generating certificate for ' + host
+        args = [
+            '/usr/bin/env', 'python', script,
+            '--account-key',
+            '{{ letsencrypt_account_key }}',
+            '--csr',
+            '{{ acme_tiny_data_directory }}/csrs/' + cert['name'] + '.csr',
+            '--acme-dir',
+            '{{ acme_tiny_challenges_directory }}'
+        ]
 
-    print "Generating certificate for " + host
-    args = [
-        "/usr/bin/env", "python", script,
+        cmd = ' '.join(args)
+        p = Popen(
+            cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=PIPE, close_fds=True,
+            preexec_fn=demote(user_uid, user_gid), env=env
+        )
+        output = p.stdout.read()
+        p.stdin.close()
+        if p.wait() != 0:
+            print 'error while generating certificate for {}'.format(host)
+            print p.stderr.read()
+        else:
+            f = open(cert['certpath'], 'w')
+            f.write(output)
+            f.close()
+            renewed = True
 
-        "--account-key",
-        "{{ letsencrypt_account_key }}",
-        "--csr",
-        "{{ acme_tiny_data_directory }}/csrs/" + cert['name'] + ".csr",
-        "--acme-dir",
-        "{{ acme_tiny_challenges_directory }}"
-    ]
+    return renewed
 
-    cmd = "/usr/bin/env " + " ".join(args)
 
-    p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=PIPE, close_fds=True)
-    output = p.stdout.read()
-    p.stdin.close()
-    if p.wait() != 0:
-        print "error while generating certificate for " + host
-        print p.stderr.read()
+def run_post_renew_cmd():
+    """Optionally runs a command after renewal (run as the root user)"""
+    post_renew_cmd = '{{ letsencrypt_post_renew_cmd }}'
+    if not len(post_renew_cmd):
+        print 'No post-renew command configured'
+        return
+
+    process = Popen(
```

Comment:
Is it possible to simplify this down to:
`retcode = subprocess.call(post_renew_cmd)`? I suspect that it might be possible to replace the call above with `subprocess.check_output()` but there may be additional considerations that apply there that aren't as relevant here so I'm inclined to leave that one as-is for now.